modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -308,23 +376,37 @@ public void onLocalJoin() {
         if (res != null)
             return res;
 
-        if (node.isClient())
+        if (isMasterKeyChangeInProgress()) {
+            // Prevents new nodes join to avoid inconsistency of the master key.
+            return new IgniteNodeValidationResult(ctx.localNodeId(),
+                "Master key change in progress! Node join is rejected. [node=" + node.id() + "]",
+                "Master key change in progress! Node join is rejected.");
+        }
+
+        if (node.isClient() || node.isDaemon())
             return null;
 
         res = validateNode(node);
 
         if (res != null)
             return res;
 
-        if (!discoData.hasJoiningNodeData()) {
-            U.quietAndInfo(log, "Joining node doesn't have encryption data [node=" + node.id() + "]");
+        NodeEncryptionKeys nodeEncKeys;
 
-            return null;
+        if (!discoData.hasJoiningNodeData() ||
+            (nodeEncKeys = (NodeEncryptionKeys)discoData.joiningNodeData()) == null) {
```

Comment:
Please, don't use the assignment operator in if.

Reply:
Done.